File: Vaishnavi/14-15/container.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maxi = 0
        while (left<right):
            h1 = height[left]
            h2 = height[right]
            mini = min(h1,h2)
            maxi = max(maxi,mini*(right-left))
            if h1<h2:
                left += 1
            else:
                right -= 1
        return maxi

File: Chandra Kiran/14-09-2022/container.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    var i =0, j = height.length-1;
    var max = 0, temp = 0;
    while(i!=j){
        temp = Math.min(height[i], height[j]) * (j - i);
        if (temp > max) {
            max = temp;
        }
        if (height[i] < height[j]) {
            i++
        }
        else {
            j--
        }
    }
    return max
};

maxArea([1, 8, 6, 2, 5, 4, 8, 3, 7])